data/filedefs/filetypes.cobol

```diff
@@ -16,7 +16,8 @@ quotedidentifier=identifier_2
+secondary=access address all alphabetic alphabetic-lower alphabetic-upper alphanumeric alphanumeric-edited any area areas argument-value asc binary blank block by char character characters column comma commarea common comp comp-0 comp-1 comp-2 comp-3 comp-4 comp-5 comp-6 comp-7 comp-8 comp-9 computational computational-0 computational-1 computational-2 computational-3 computational-4 computational-5 computational-6 computational-7 computational-8 computational-9 content copy current cursor date date-compiled date-written day day-of-week decimal decimal-point desc distinct dynamic environment-value equal external f false file filler global greater high-value high-values hold id in include index indexed input integer key length label less low-value low-values mode modules null nulls number numeric numeric-edited occurs of omitted only organization other output packed-decimal page page-counter pointer procedures queue quote quotes random record records reference references row rows s9 screen size smallint space spaces sqlca sqlcode standard standard-1 standard-2 status sub-queue-1 sub-queue-2 sub-queue-3 table than time times timestamp true up value values words x xx xxx xxxx xxxxx xxxxxx z z9 zz zz9 zzz zzz9 zzzz zzzz9 zzzzz zzzzz9 zzzzzz zzzzzz9 zzzzzzz zzzzzzz9 zzzzzzzz zzzzzzzz9 zzzzzzzzz zzzzzzzzz9 zzzzzzzzzz zzzzzzzzzz9 zzzzzzzzzzz zzzzzzzzzzz9 zzzzzzzzzzzz zzzzzzzzzzzz9 zzzzzzzzzzzzz zzzzzzzzzzzzz9 zzzzzzzzzzzzzz zzzzzzzzzzzzzz9 zzzzzzzzzzzzzzz zzzzzzzzzzzzzzz9 zero zeroes zeros
```

Comment:
Hi,

I started with cobol.vim by Tim Pope but main of the adds were by knowledge, use and experience.

Trust me, I didn't find a proper COBOL definition since a long time but every word I add is verified.

For exemple, RENAMES you could find this > http://www.techtricky.com/cobol-renames-clause-with-examples/ or by IBM > https://www.ibm.com/support/knowledgecenter/SSAE4W_9.1.0/com.ibm.etools.iseries.langref.doc/evfeb4ls37.htm

For DECLARATIVES (by IBM)> https://www.ibm.com/support/knowledgecenter/SSPSQF_9.0.0/com.ibm.etools.iseries.langref.doc/c0925395314.htm

> x xx xxx xxxx xxxxx xxxxxx z z9 zz zz9 zzz zzz9 zzzz zzzz9 zzzzz zzzzz9 zzzzzz zzzzzz9 zzzzzzz zzzzzzz9 zzzzzzzz zzzzzzzz9 zzzzzzzzz zzzzzzzzz9 zzzzzzzzzz zzzzzzzzzz9 zzzzzzzzzzz zzzzzzzzzzz9 zzzzzzzzzzzz zzzzzzzzzzzz9 zzzzzzzzzzzzz zzzzzzzzzzzzz9 zzzzzzzzzzzzzz zzzzzzzzzzzzzz9 zzzzzzzzzzzzzzz zzzzzzzzzzzzzzz9

Above it is for the declaration of variables in characters or numeric. The 'x' ones are for some characters declarations and 'z' ones for numbers. I know that these types of statements may seem strange but I have not found other ways to translate them and here all cases are covered

For example you could have `05 WS-EXCEEDS-PTS-P    PIC ZZZZZZZZ9.99 VALUE ZEROS.`
Or `05 W-SIT-ISC-IMS-SSID         PIC XXXX.`

Reply:
aren't the zzzzzzzzzzzzzzzz things formatting, not keywords?

Reply:
> aren't the zzzzzzzzzzzzzzzz things formatting, not keywords?

It's PIC clause, it's slightly different ;')

PS : For COMPUTATIONAL (still by IBM) > https://www.ibm.com/support/knowledgecenter/fr/SSKPEG_9.5.0/com.ibm.pdp.pac.doc/topics/rsegment_ce_fillerdef.html

Reply:
The link points to something about an unintelligible language (Cobol) in an unintelligible language (French) so its no help.

PIC effectively is the record format, but however you want to look at it, the point is its _not_ a keyword.

Reply:
> The link points to something about an unintelligible language (Cobol) in an unintelligible language (French) so its no help.

My bad, I think I put an english link but I wrongly connect with my accunt before to copy, here the good one > https://www.ibm.com/support/knowledgecenter/en/SSKPEG_9.5.0/com.ibm.pdp.pac.doc/topics/rsegment_ce_fillerdef.html

> PIC effectively is the record format, but however you want to look at it, the point is its _not_ a keyword.

Yes certainly, it's just more readable like this but you are master on that

Reply:
Do I have to erase them from filedef ?

Reply:
> For exemple, RENAMES you could find this > http://www.techtricky.com/cobol-renames-clause-with-examples/ or by IBM > https://www.ibm.com/support/knowledgecenter/SSAE4W_9.1.0/com.ibm.etools.iseries.langref.doc/evfeb4ls37.htm

My point is that your version does *not* have `RENAMES`, which I found odd.  It's the same with the other ones I listed in the same bunch: they appear in the three sources I found, but not in your list.

> PS : For COMPUTATIONAL (still by IBM) > https://www.ibm.com/support/knowledgecenter/fr/SSKPEG_9.5.0/com.ibm.pdp.pac.doc/topics/rsegment_ce_fillerdef.html

I've seen `COMPUTATIONAL` up to 6, but your link lists 7 and 9, so OK.

> Do I have to erase them from filedef ?

I don't have a strong opinion as I don't know COBOL, but I'd rather separate this.  Maybe put it in `extended_keywords` so it's at least in a different list?

BTW, how did you split secondary and primary?  Just curious.

Reply:
> My point is that your version does not have RENAMES, which I found odd. It's the same with the other ones I listed in the same bunch: they appear in the three sources I found, but not in your list.

It's true. I mainly add the reserved words I see often and those of the Tim Pope's list as I said. If you're OK I'll have a look within a few days and let you know.

> I don't have a strong opinion as I don't know COBOL, but I'd rather separate this. Maybe put it in extended_keywords so it's at least in a different list?

OK I'll also test it and let you know. Agree ?

> BTW, how did you split secondary and primary? Just curious.

I simply tried to get as close as possible to what you could find on the IBM editor and made some adaptations for more readability

Reply:
Sure, sounds great :)